modules/setting/setting.go

```diff
@@ -208,17 +205,29 @@ func PrepareAppDataPath() error {
 
 // InitProviderFromExistingFile initializes config provider from an existing config file (app.ini)
 func InitProviderFromExistingFile() {
-	CfgProvider = newFileProviderFromConf(CustomConf, false, "")
+	var err error
+	CfgProvider, err = newConfigProviderFromFile(CustomConf, false, "")
+	if err != nil {
+		log.Fatal("%v", err)
+	}
 }
 
 // InitProviderAllowEmpty initializes config provider from file, it's also fine that if the config file (app.ini) doesn't exist
 func InitProviderAllowEmpty() {
-	CfgProvider = newFileProviderFromConf(CustomConf, true, "")
+	var err error
+	CfgProvider, err = newConfigProviderFromFile(CustomConf, true, "")
+	if err != nil {
+		log.Fatal("%v", err)
+	}
 }
 
 // InitProviderAndLoadCommonSettingsForTest initializes config provider and load common setttings for tests
 func InitProviderAndLoadCommonSettingsForTest(extraConfigs ...string) {
-	CfgProvider = newFileProviderFromConf(CustomConf, true, strings.Join(extraConfigs, "\n"))
+	var err error
+	CfgProvider, err = newConfigProviderFromFile(CustomConf, true, strings.Join(extraConfigs, "\n"))
+	if err != nil {
+		log.Fatal("%v", err)
+	}
```

Comment:
I merged two of them, the third cannot be merged easily.

Reply:
I didn't do that because the code  `LoadXxx(true)` and `LoadXxx(false)` are really difficult to read.

At least, we should do this:

```
LoadXxxFromExisting()  { loadXxx(true) }
LoadXxxAllowEmpty() { loadXxx(false) }
```

Reply:
This is what the code looks like previously.